app/models/badge.rb

```diff
@@ -0,0 +1,60 @@
+class Badge < ApplicationRecord
+
+  has_many :achievements, dependent: :destroy
+  has_many :users, through: :achievements
+  has_one_attached :image
+
+  validates :title, presence: true
+  validates :title, uniqueness: true
+  validates :description, presence: true
+
+  def conditions
+    [
+      [ "Пройти все тесты категории: ", "category_complited" ],
+      [ "Пройти все тесты уровня: ", "level_complited" ],
+      [ "Сдать первый тест", "first_success" ],
+      [ "Завалить первый тест", "first_fail" ]
+    ]
+  end
+
+  def check(result)
+    send(condition, result)
+  end
+
+  private
+
+  def first_success(result)
+    id if (result.user.results.count <= 1) && result.pass?
+  end
+
+  def first_fail(result)
+    id if (result.user.results.count <= 1) && (result.pass? == false)
+  end
+
+  def category_complited(result)
+    category_id = additional_condition.to_i
+    return nil if result.test.category_id != category_id
```

Comment:
```suggestion
    return if result.test.category_id != category_id
```
`return` по дефолту возвращает `nil`